### PR TITLE
feat: Installation access tokens for Git operations on GH PLUTO-772

### PR DESCRIPTION
**Pull request by claudiacarpinteiro**
For GitHub organizations, the Codacy GitHub App will start [authenticating as an app installation](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation) and using installation access tokens for Git operations.

This pull request updates the documentation to consider this product change.

### :construction: To do
-   [ ] Update content considering the new behavior
-   [ ] Request validation
-   [ ] Apply feedback
-   [ ] Wait for feature release

### CODE REVIEW

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 1]
```diff
@@ -63,15 +63,10 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
       <td>Read & Write</td>
       <td>Codacy sets the status of commits according to the result of code analysis.</td>
     </tr>
-    <tr>
-      <td>Administration</td>
```
Comment (claudiacarpinteiro) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:67]: @andrzej-janczak Using installation access tokens on GitHub, I believe we don't need the `Administration` permission anymore. Can you please confirm? 🙏

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 2]
```diff
@@ -206,8 +201,13 @@ If you need to use an integration that you have previously revoked, log in again
...
+On GitLab and Bitbucket organizations, when you add a private repository to Codacy, Codacy uses the integration with your Git provider to create a new SSH key on the repository. Codacy then uses that SSH key every time it needs to clone the repository.
+
+**Codacy only adds read-only SSH keys** and can't access any of your existing SSH keys. You have full control over which organizations and repositories Codacy is authorized to access. Codacy doesn't change the contents or member privileges of any repository you authorize it to analyze.
 
 We understand the desire for security and privacy and find that the SSH protocol is preferable to HTTPS as it separates Codacy's access rights from the one of the users.
+
+!!! tip
```
Comment (claudiacarpinteiro) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:212]: @andrzej-janczak Can you please validate if these are the most indicated links to GitLab and Bitbucket documentation?

File: docs/repositories-configure/using-submodules.md [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 By default, Codacy does normal Git clones that **don't include submodules** to ensure that we only clone necessary repositories. If your organization needs to use submodules, you can request Codacy to enable this feature for you.
 
+!!! note
+    On GitHub organizations, your repository and the repositories that you add as submodules must belong to the same GitHub organization.
```
Comment (claudiacarpinteiro) [inline docs/repositories-configure/using-submodules.md:8]: @andrzej-janczak I added the new limitation for GH when using submodules. Let me know if this sentence is clear. 🙏

File: docs/repositories-configure/using-submodules.md [hunk 4]
```diff
@@ -4,35 +4,50 @@
...
+
 ## Prerequisites for using submodules
 
 1.  Contact us at <mailto:support@codacy.com> asking to enable submodules on Codacy.
 
 1.  **If you're using Codacy Self-hosted**, [update your license](../chart/maintenance/license.md).
 
-1.  If your submodules are:
+1.  If your submodules are:<!--TODO PLUTO-772 Validate prerequisites-->
```
Comment (claudiacarpinteiro) [inline docs/repositories-configure/using-submodules.md:16]: @andrzej-janczak Now using installation access tokens, do these prerequisites related to the repository Git URL still apply to GitHub? 🤔

File: docs/repositories-configure/using-submodules.md [hunk 5]
```diff
@@ -4,35 +4,50 @@
...
+1.  If you're using submodules to share an analysis tool configuration file across your repositories, check if your tool recursively searches the subdirectories of your repositories for configuration files.
 
-1.  For GitHub and Bitbucket, remove this Codacy key from the repository settings on your Git provider.
+    If your tool doesn't detect the configuration files in the submodule directories, you must include a configuration file directly in the root of your repositories referencing the configuration files in the submodule directories.
 
-1.  Add a new SSH key to your git provider account by clicking the link **Add new user key** or the button **Generate New User Key**, depending on your Git provider.
+## Updating the public SSH key to access the repository {: id="update-key"}
 
-    For GitHub and Bitbucket, this takes you to the Git provider page where you can manage your user account SSH keys. For GitLab, Codacy removes the existing repository key and creates the new SSH key on your user account automatically.
+!!! info "This section applies only to GitLab and Bitbucket"
```
Comment (nicklem) [inline docs/repositories-configure/using-submodules.md:32]: If the entire section applies only to specific providers, I agree that an admonition is best 👍

File: docs/repositories-configure/using-submodules.md [hunk 6]
```diff
@@ -4,35 +4,50 @@
...
-    If your tool doesn't detect the configuration files in the submodule directories, you must include a configuration file directly in the root of your repositories referencing the configuration files in the submodule directories.
+1.  Open the repository **Settings**, tab **General**. In the **Danger zone** area, you have the **SSH Key** generated by Codacy to access your repository.
+
+1.  Depending on your Git provider, do the following to update the key:
+
+    -   For GitLab, click the button **Generate New User Key**. Codacy removes the existing repository key and creates the new SSH key on your user account automatically.
+
+    -   For Bitbucket:
+        1.  Take note of the existing key.
+        1.  Remove this Codacy key from the repository settings on your Git provider.
```
Comment (nicklem) [inline docs/repositories-configure/using-submodules.md:46]: Could we simplify here?
```suggestion
        1.  Remove the existing Codacy key from the repository settings on your Git provider.
```

**Review by nicklem — COMMENTED** [review]
Looking good! Left some suggestions.

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 1 above)
Comment (andrzej-janczak) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:67]: Yes, this is true, for sure we drop `write` access 
read access is to be checked in task: https://codacy.atlassian.net/browse/PLUTO-765 
but for 99% will be dropped also, so  I for now I would assume we just drop `Administration`

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 2 above)
Comment (andrzej-janczak) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:212]: hm for GL we use deploy keys that are located here
doc: https://docs.gitlab.com/ee/user/project/deploy_keys/index.html#view-deploy-keys

Screenshot 2023-11-14 at 09 52 54

for BB looks like the best place would be: 
https://support.atlassian.com/bitbucket-cloud/docs/configure-repository-settings/ 
Screenshot 2023-11-14 at 09 55 23

this is also like a deploy key (read only ssh key per repo) but not using "deploy" word.

File: docs/repositories-configure/using-submodules.md (hunk 3 above)
Comment (andrzej-janczak) [reply docs/repositories-configure/using-submodules.md:8]: yes, this is true  👍

File: docs/repositories-configure/using-submodules.md (hunk 4 above)
Comment (andrzej-janczak) [reply docs/repositories-configure/using-submodules.md:16]: For github there will be no SSH at all 

Thus for both private and public for Github it should use https

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 1 above)
Comment (claudiacarpinteiro) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:67]: Ok. Let me know when we finish that task. If we keep `Administration` Read permission, I will need to explain how we use that permission.

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 2 above)
Comment (claudiacarpinteiro) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:212]: Updated!

File: docs/repositories-configure/removing-your-repository.md [hunk 7]
```diff
@@ -16,4 +16,4 @@ To delete your repository from Codacy:
     ![Removing your repository](images/repository-remove.png)
 
     !!! note
-        For added security, after you remove the repository from Codacy you can delete the webhooks and SSH keys related to this Codacy repository from your Git provider to prevent their reuse.
+        For added security, after you remove the repository from Codacy you can delete the webhooks and SSH keys (GitLab and Bitbucket only) related to this Codacy repository from your Git provider to prevent their reuse.
```
[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/removing-your-repository.md:19]: The recommendation to delete webhooks applies to all three providers, the limitation applies only to SSH keys. Maybe it's not clear enough?

Adding an alternative below, but I think the sentence got more complex... 🤔 
Thought? 

```suggestion
        For added security, after you remove the repository from Codacy you can delete from your Git provider the resources related to this Codacy repository to prevent their reuse:
        -   Webwooks
        -   SSH keys (GitLab and Bitbucket only)
```

File: docs/repositories-configure/using-submodules.md [hunk 8]
```diff
@@ -4,35 +4,50 @@
...
+1.  If your submodules are:<!--TODO PLUTO-772 Validate prerequisites-->
     -   **Public repositories**, make sure that your Git URL uses the HTTPS protocol.
     -   **Private repositories**, make sure that your Git URL uses the SSH protocol.
 
 ## Enabling submodules on a repository
 
 When using submodules, you must do the following for all your existing and new repositories:
 
-1.  Open the repository **Settings**, tab **General**. In the **Danger zone** area, you have the **SSH Key** generated by Codacy to access your repository. Take note of this key.
+1.  **For GitLab and Bitbucket**, [update the public SSH key](#update-key) that Codacy uses to access your repository.
```
[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/using-submodules.md:26]: I'll keep the bold formatting. 😄 

```suggestion
1.  **GitLab and Bitbucket only:** [Update the public SSH key](#update-key) that Codacy uses to access your repository.
```

File: docs/repositories-configure/removing-your-repository.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (nicklem) [reply docs/repositories-configure/removing-your-repository.md:19]: True, the sentence is a bit more complex, but it's much clearer that webhooks apply to all three. Looks better to me like this, yes 👍

[the remark this answers is not in the document]
Comment (claudiacarpinteiro) [reply docs/repositories-configure/removing-your-repository.md]: Something went wrong with this commit, I think I committed the wrong suggestion. Fixing this manually.

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 1 above)
Comment (claudiacarpinteiro) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:67]: @andrzej-janczak I kept the Administration permission for now describing the fallback mechanism. Let me know if it's correct.

Comment (andrzej-janczak) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:67]: LGTM

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 9]
```diff
@@ -63,15 +63,15 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
...
     <tr>
       <td>Contents</td>
       <td>Read-Only</td>
-      <td>Codacy accesses repository contents to provide faster code coverage analysis and as part of an initiative to use GitHub App tokens instead of SSH keys when cloning repositories for code quality analysis.</td>
+      <td>Codacy retrieves repository contents when integrating with your repositories and cloning them for code quality analysis, and to provide faster code coverage analysis.</td>
+    </tr>
+    <tr>
+      <td>Administration</td>
+      <td>Read & Write</td>
+      <td>This permission will soon be removed and is currently used <strong>exclusively as a fallback mechanism</strong> when the Contents permission is missing. In this case, Codacy <a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">creates an SSH key on the repository</a> to allow cloning and integrating with your repository.<br/>To avoid disruptions in the future, make sure you accept Codacy GitHub App updated permissions on your GitHub organization.</td>
```
Comment (andrzej-janczak) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:74]: maybe we could add link to GH docs 
https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 10]
```diff
@@ -63,15 +63,15 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
...
     <tr>
       <td>Contents</td>
       <td>Read-Only</td>
-      <td>Codacy accesses repository contents to provide faster code coverage analysis and as part of an initiative to use GitHub App tokens instead of SSH keys when cloning repositories for code quality analysis.</td>
+      <td>Codacy retrieves repository contents when integrating with your repositories and cloning them for code quality analysis, and to provide faster code coverage analysis.</td>
+    </tr>
+    <tr>
+      <td>Administration</td>
+      <td>Read & Write</td>
+      <td>This permission will soon be removed and is currently used <strong>exclusively as a fallback mechanism</strong> when the Contents permission is missing. In this case, Codacy <a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">creates an SSH key on the repository</a> to allow cloning and integrating with your repository.<br/>To avoid disruptions in the future, make sure you [approve Codacy GitHub App updated permissions](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization.</td>
```
Comment (claudiacarpinteiro) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:74]:
```suggestion
      <td>This permission will soon be removed and is currently used <strong>exclusively as a fallback mechanism</strong> when the Contents permission is missing. In this case, Codacy <a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">creates an SSH key on the repository</a> to allow cloning and integrating with your repository.<br/>To avoid disruptions in the future, make sure you <a href="https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app">approve Codacy GitHub App updated permissions</a> on your GitHub organization.</td>
```

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 11]
```diff
@@ -63,15 +63,15 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
...
-    <tr>
-      <td>Administration</td>
-      <td>Read & Write</td>
-      <td><a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">Codacy creates an SSH key</a> on the repository to allow cloning and integrating with your repository.</td>
-    </tr>
     <tr>
       <td>Contents</td>
       <td>Read-Only</td>
-      <td>Codacy accesses repository contents to provide faster code coverage analysis and as part of an initiative to use GitHub App tokens instead of SSH keys when cloning repositories for code quality analysis.</td>
+      <td>Codacy uses installation access tokens when integrating with your repositories and cloning them for code quality analysis, and retrieves repository contents to provide faster code coverage analysis.</td>
```
Comment (claudiacarpinteiro) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: @andrzej-janczak @lorandszakacs I did a tweak here to mention **installation access tokens**. Let me know if it's correct. 🙏 
I want to confirm if Codacy **uses** installation access tokens. (instead of creating/generating... for that maybe we would need more than read-only access, right?)

Comment (andrzej-janczak) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: look ok 👍

**Review by andrzej-janczak — APPROVED** [review]

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 11 above)
Comment (nicklem) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: Not sure I'm getting the meaning of this sentence. I'm attempting a reword, let me know if it's correct:

```suggestion
      <td>Codacy uses installation access tokens when integrating with your repositories to clone them for code quality analysis and to retrieve their contents for faster code coverage analysis.</td>
```

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 12]
```diff
@@ -206,8 +206,14 @@ If you need to use an integration that you have previously revoked, log in again
 
 ## Why does Codacy ask for permission to create SSH keys?
 
-When you add a private repository to Codacy, Codacy uses the integration with your Git provider to create a new SSH key on the repository. Codacy then uses that SSH key every time it needs to clone the repository.
+!!! info "This section applies only to GitLab and Bitbucket"
+    For GitHub, Codacy creates an SSH key on the repository **exclusively as a fallback mechanism** when the Contents permission is missing. To avoid disruptions in the future, make sure you accept Codacy GitHub App [updated permissions](#github-cloud) on your GitHub organization.
```
Comment (nicklem) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:210]: Does this admonition break our usual structure? I remember using custom titles only with empty `info` admonitions.

Either way, I'm OK with using it like this here. This is temporary I presume?

File: docs/repositories-configure/using-submodules.md [hunk 13]
```diff
@@ -4,35 +4,54 @@
 
 By default, Codacy does normal Git clones that **don't include submodules** to ensure that we only clone necessary repositories. If your organization needs to use submodules, you can request Codacy to enable this feature for you.
 
+!!! important
+    **GitHub only:**
+
+    -   To clone repositories, Codacy GitHub App requires Contents permission. Make sure you accept Codacy GitHub App [updated permissions](../getting-started/which-permissions-does-codacy-need-from-my-account.md/#github-cloud) on your GitHub organization.
```
Comment (nicklem) [inline docs/repositories-configure/using-submodules.md:10]: Nitpick: adding "the" for flow:

```suggestion
    -   To clone repositories, the Codacy GitHub App requires the Contents permission. Make sure you accept the Codacy GitHub App [updated permissions](../getting-started/which-permissions-does-codacy-need-from-my-account.md/#github-cloud) on your GitHub organization.
```

**Review by nicklem — COMMENTED** [review]
LGTM! Left a couple nitpicks.

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 11 above)
Comment (claudiacarpinteiro) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: > Codacy uses installation access tokens when integrating with your repositories to clone them for code quality analysis and to retrieve their contents for faster code coverage analysis.

@nicklem This way we are saying that the new Coverage engine also uses installation access tokens... Does it use tokens? 🤔 
I based on the [original sentence](https://github.com/codacy/docs/pull/1931/files#diff-47d9d57a1ef035b843d334b8ceabbfee55e41dc6fe3e45012197b57631db1c28L74), which does not specify if Contents permissions is required to use tokens or something else...

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 12 above)
Comment (claudiacarpinteiro) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:210]: You are right, this is the first time we add additional content to an `info` admonition. But at the same time, we always use `info` admonitions to say "This page/section only applies to <something>"...

So I opted to break our usual standard by adding more information (because it's temporary), instead of breaking by using a different admonition type. 🤓 

I would keep it like this, as we'll be removing the additional temporary content in short/medium future.

WDYT?

Comment (nicklem) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:210]: Sounds good, especially if this is temporary 👍

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 11 above)
Comment (nicklem) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: Oh, I see. I got the meaning wrong. The fact is that the fragment at the beginning, "Codacy uses installation access tokens", seems to imply that the installation access tokens are used to access contents, but it's the other way around, i.e. the Contents permission is used to access the installation access tokens?

If so, maybe rewording like this can work?

Either:
```
 <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code quality analysis, and to provide faster code coverage analysis.</td>
```

Or:
```
<td>Codacy retrieves repository contents to provide faster code coverage analysis and to retrieve installation access tokens, which are used to integrate with your repositories and clone them for code quality analysis</td>
```

Comment (nicklem) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: I wonder if we could simplify further, avoid mentioning "faster code coverage analysis", and simply group that into a generic "clone them for code analysis":

```
<td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```

Comment (nicklem) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: I wonder if we could simplify further, avoid mentioning "faster code coverage analysis", and simply group that into a generic "clone them for code analysis":

```suggestion
      <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```

Comment (nicklem) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: Perhaps we can simplify even further:

```suggestion
      <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 14]
```diff
@@ -63,15 +63,15 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
...
     <tr>
       <td>Contents</td>
       <td>Read-Only</td>
-      <td>Codacy accesses repository contents to provide faster code coverage analysis and as part of an initiative to use GitHub App tokens instead of SSH keys when cloning repositories for code quality analysis.</td>
+      <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
+    </tr>
+    <tr>
+      <td>Administration</td>
+      <td>Read & Write</td>
+      <td>This permission will soon be removed and is currently used <strong>exclusively as a fallback mechanism</strong> when the Contents permission is missing. In this case, Codacy <a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">creates an SSH key on the repository</a> to allow cloning and integrating with your repository.<br/>To avoid disruptions in the future, make sure you <a href="https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app">approve Codacy GitHub App updated permissions</a> on your GitHub organization.</td>
```
Comment (jorgebraz) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:74]: I would suggest the emphasis (strong) in this text should be on _the soon to be deprecated permission_, rather than the current use we have for it.
ex: `<strong>This permission will soon be removed</strong> and is currently used exclusively as a fallback mechanism (...)`

Also, as we are still figuring out possible issues and ironing out details as we go, it's gonna be inherently difficult to align a "release date" for this, so I'd suggest we adapt language to accommodate for this, something like avoiding the term "exclusively": 
> This permission will soon be removed and is currently used when it's not possible to leverage the Repository Contents permission. In this case, Codacy creates an SSH key on the repository to allow cloning and integrating with your repository.
> To future-proof your codacy usage, make sure you that you aprove [approve Codacy GitHub App updated Repository Coontents permission](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization.

It's possible that I'm being creative with the usage of "not possible", I was trying, without lying to encapsulate both the cases that we reverted the behaviour for some testing reason as well as the cases where customers have not accepted the Repository `Contents` permission.

(Also please don't use any of my suggestions verbatim, I'm sure there are better ways to write these things.)

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md [hunk 15]
```diff
@@ -63,15 +63,15 @@ If you log in with GitHub, Codacy requires the following [app permissions](https
...
-    <tr>
-      <td>Administration</td>
-      <td>Read & Write</td>
-      <td><a href="#why-does-codacy-ask-for-permission-to-create-ssh-keys">Codacy creates an SSH key</a> on the repository to allow cloning and integrating with your repository.</td>
-    </tr>
     <tr>
       <td>Contents</td>
       <td>Read-Only</td>
-      <td>Codacy accesses repository contents to provide faster code coverage analysis and as part of an initiative to use GitHub App tokens instead of SSH keys when cloning repositories for code quality analysis.</td>
+      <td>Codacy retrieves repository contents to get installation access tokens, which are used to integrate with your repositories and clone them for code analysis.</td>
```
Comment (jorgebraz) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:69]: I believe we could also include `code coverage` in the first line (I'll be honest, I don't fully understand how and why the new coverage pipeline needs this permission, but it does)

I'd also suggest adding an emphasized call to action here for people to approve this permission.
> Codacy is planing on deprecating the need for Repository Administration permissions, but to do so resources from Contents permssion are needed. This Permission has strated being requesteted recently so to future-proof your codacy usage, make sure you that you aprove [approve Codacy GitHub App updated Repository Coontents permission](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization.

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 12 above)
Comment (jorgebraz) [inline docs/getting-started/which-permissions-does-codacy-need-from-my-account.md:210]: > For Github, Codacy only uses an SSH key when it's not possible to leverage the Repository Contents permission. In this case, Codacy uses a repository SSH key allow cloning and integrating with your repository. As we progress with our plan to deprecate Repository Administration permissions we will fully stop to create, use and support SSH keys. To avoid disruptions in the future, make sure you accept Codacy GitHub App [updated permissions](#github-cloud) on your GitHub organization.

`Codacy creates an SSH key on the repository` this sentence is not true yet. SSH keys are created when a repository is added, and we will only stop this behaviour when we fully deprecate the `Administration` permission.

File: docs/getting-started/which-permissions-does-codacy-need-from-my-account.md (hunk 14 above)
Comment (claudiacarpinteiro) [reply docs/getting-started/which-permissions-does-codacy-need-from-my-account.md]: > It's possible that I'm being creative with the usage of "not possible", I was trying, without lying to encapsulate both the cases that we reverted the behaviour for some testing reason as well as the cases where customers have not accepted the Repository Contents permission.

I rephrased it according to the current state of the art. 😉

**Comment by claudiacarpinteiro** [comment]
@jorgebraz Thank you for the input. I incorporated your feedback. 🙏
